HACKER/HACKER/Sources/ViewControllers/MainViewController/MainProfileViewController.swift

```diff
@@ -0,0 +1,108 @@
+//
+//  MainProfileViewController.swift
+//  HACKER
+//
+//  Created by 김지수 on 2022/03/18.
+//
+
+import UIKit
+
+import SnapKit
+import Then
+
+// MARK: - MainProfileViewController
+class MainProfileViewController: UIViewController {
+  
+  // MARK: - Components
+  let backButton = UIButton()
+  let userCharacterImage = UIImageView()
+  let userHairImage = UIImageView()
+  let userNicknameLabel = UILabel()
+  let userGithubNameLabel = UILabel()
+  let hairNumLabel = UILabel()
+  
+  // MARK: - LifeCycle
+  override func viewDidLoad() {
+    super.viewDidLoad()
+    self.view.backgroundColor = .hackerWhite
+    self.navigationController?.navigationBar.isHidden = true
+    layout()
+  }
+}
+// MARK: - Extensions
+extension MainProfileViewController {
+  func layout() {
+    layoutBackButton()
+    layoutUserCharacterImage()
+    layoutUserHairImage()
+    layoutUserNickNameLabel()
+    layoutUserGithubNameLabel()
+    layoutHairNumLabel()
+  }
+  func layoutBackButton() {
+    view.add(backButton) {
+      $0.setImage(UIImage(named: "backButtonIcon"), for: .normal)
+      $0.addTarget(self, action: #selector(self.backButtonTapped), for: .touchUpInside)
+      $0.snp.makeConstraints {
+        $0.top.equalTo(self.view.safeAreaLayoutGuide.snp.top).offset(16)
+        $0.leading.equalToSuperview().offset(24)
+        $0.width.equalTo(40)
+        $0.height.equalTo(40)
+      }
+    }
+  }
+  func layoutUserCharacterImage() {
+    view.add(userCharacterImage) {
+      $0.image = UIImage(named: "userCharacterImage")
+      $0.contentMode = .scaleAspectFit
+      $0.snp.makeConstraints {
+        $0.top.equalTo(self.backButton.snp.bottom).offset(77)
+        $0.centerX.equalToSuperview()
+        $0.leading.equalToSuperview().offset(24)
+        $0.height.equalTo((UIScreen.main.bounds.width-48)*1.1)
+      }
+    }
+  }
+  func layoutUserHairImage() {
+    view.add(userHairImage) {
+      $0.image = UIImage(named: "sampleHairImage")
+      $0.contentMode = .scaleAspectFit
+      $0.snp.makeConstraints {
+        $0.top.equalTo(self.userCharacterImage.snp.top)
+        $0.centerX.equalToSuperview()
+        $0.leading.equalToSuperview().offset(24)
+        $0.height.equalTo((UIScreen.main.bounds.width-48)*1.1)
+      }
```

Comment:
```suggestion
      $0.snp.makeConstraints {
        $0.top.equalTo(self.userCharacterImage.snp.top)
        $0.centerX.equalToSuperview()
        $0.leading.equalToSuperview().offset(24)
        $0.height.equalTo(UIScreen.main.bounds.width-48).multipliedBy(1.1)
      }
```
SnapKit을 만든 사람이 스토리보드에서 레이아웃 잡아주던 거를 코드로 옮겨올 때 그 형식을 그대로 옮겨오길 원했어
그래서 equalTo 안에 스토리보드에 쓸 수 없는 수식을 쓰는 것보다 equalTo와 multipliedBy를 분리해주는 게 좀 더 직관적인 느낌이야
스토리보드에서 constant랑 multiplier를 지정하던 방식이랑 똑같이 가는 방법!

<br/>

```suggestion
      $0.snp.makeConstraints { make in
        make.top.equalTo(self.userCharacterImage.snp.top)
        make.centerX.equalToSuperview()
        make.leading.equalToSuperview().offset(24)
        make.height.equalTo(UIScreen.main.bounds.width-48).multipliedBy(1.1)
      }
```
그리고 이건 여담인데, 또 그 SnapKit을 만든 사람이 코드를 직관적으로 쓰게 하기 위해서 영어 문장 형식으로 읽히게끔 만들어놨거든
그래서 보통 make를 써줘서 문장 형식을 맞춰주는 편이야
읽어보면 Make centerX equal to superview (centerX를 superview와 같게 해라) 이렇게 읽힐 수 있게 만든 라이브러리야